src/utils/signals.c

```diff
@@ -6,17 +6,55 @@
 /*   By: aulicna <aulicna@student.42.fr>            +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/11/23 14:33:13 by aulicna           #+#    #+#             */
-/*   Updated: 2024/01/12 09:39:59 by aulicna          ###   ########.fr       */
+/*   Updated: 2024/01/17 12:43:27 by aulicna          ###   ########.fr       */
```

Comment:
The documentation makes most sense when read in the normal program flow (minishell_loop > handle_sigint > heredoc files > handle_sigint_heredoc > reset in minishell_loop).